### ISSUE TITLE
Rendering a new View.Grid displaces content inside

### ISSUE THREAD

**Issue by rillstone** [fsprojects/Fabulous#744]
When switching from one `View.Grid` to another, it seems to displace the content inside as it reuses the attributes such as the `coldefs` and `rowdefs` from the previous grid. 
I believe this could be a similar issue to #622 which was fixed in PR #627. However, I am experiencing similar behaviour to this issue now. 
I am using Fabulous `0.54.0` and Xamarin.Forms `4.6.0.726`

Reproducing this issue with:
```f#
let view (model: Model) dispatch =
    match model.Page with
    | One ->
        View.ContentPage(
            content = View.Grid(
                coldefs = [Star; Stars 25.; Star],
                rowdefs = [Auto; Auto],
                children = [
                    yield View.Label(text = "Page One").Row(0).Column(1)
                    yield View.Button(text = "Change", command = (fun () -> dispatch RenderTwo)).Row(1).Column(1)
                ]
            )
        )
    | Two ->
        View.ContentPage(
            content = View.Grid(
                rowdefs = [Auto; Auto],
                children = [
                    yield View.Label(text = "Page Two").Row(0)
                    yield View.Button(text = "Change", command = (fun () -> dispatch RenderOne)).Row(1)
                ]
            )
        )
```
When clicking _Change_ from _Page One_ to _Page Two_, the content of the `View.Grid` in the second view shows as
<img src="https://user-images.githubusercontent.com/17172036/82287992-9b507500-99f5-11ea-94e2-b682de506101.jpg" width="800">

I understand that the reason for this is when updating the views, the views are being reused from the [`canReuseView` function](https://github.com/fsprojects/Fabulous/blob/d5a2e8b59d1a3e9267dfcee43455b12ae0540984/Fabulous.XamarinForms/src/Fabulous.XamarinForms.Core/ViewHelpers.fs#L16)
I worked around this issue by adding a unique `automationId` to both grids, in order to get a false return in `canReuseView` for the [`canReuseAutomationId prevChild newChild` condition](https://github.com/fsprojects/Fabulous/blob/d5a2e8b59d1a3e9267dfcee43455b12ae0540984/Fabulous.XamarinForms/src/Fabulous.XamarinForms.Core/ViewHelpers.fs#L17):
 
```f#
| One ->
    View.ContentPage(
      content = View.Grid(
          automationId="PageOne",
          coldefs = [Star; Stars 25.; Star],
          rowdefs = [Auto; Auto],
          children = [
              yield View.Label(text = "Page One").Row(0).Column(1)
              yield View.Button(text = "Change", command = (fun () -> dispatch RenderTwo)).Row(1).Column(1)
          ]
      )
    )
```
I'm not quite satisfied with this workaround as the `automationId` should be used for automation testing. Is this expected behaviour from the view reuse logic? or could there be some missing checks/edge cases inside this `canReuseView` function causing views to be reused when they shouldn't be? 

TIA

**Comment by TimLariviere** [comment]
@rillstone Thanks for the report!
`v0.54` seems ok. I don't reproduce with the sample you provided.

But I do reproduce in `0.55-preview`. Will try to fix that.

In the meantime, please use `v0.54` instead.

**Comment by TimLariviere** [comment]
Should be fixed in `0.55.0-preview3`
